.github/workflows/build-test-publish.yml

```diff
@@ -142,8 +174,8 @@ jobs:
             exit 1
           fi
 
-  publish:
-    runs-on: ubuntu-22.04
+  publish-images:
+    runs-on: ${{ matrix.arch == 'arm64' && 'pub-hk-ubuntu-22.04-arm-large' || 'ubuntu-22.04' }}
```

Comment:
The `publish-images` step is only docker push/tagging existing images. As such, it seems it can run on any runner, and so the `runs-on` property here could be `ubuntu-22.04` for all permutations in the matrix (including the arch-arm64 jobs)? This would save having to do the manual Docker install, plus mean being able to use the warm `ubuntu-22.04` runner pool GitHub has, instead of the slower org ~just-in-time-provisioned instances.

Reply:
This step is also loading images into the daemon and tagging images in the daemon. Linux docker's daemon doesn't handle storage/retrieval of cross-architecture images well (at all?).

If this step were retagging images already in a remote registry OR we were using Docker Desktop, we wouldn't need to match architectures.

Reply:
Ah right, sounds like another thing that works fine on Docker Desktop on macOS (even without the containerd snapshotter enabled), but that has rough edges with the Linux docker daemon. In which case, ignore this then :-)